portal-api/api-tests/v1/users/validate-username-and-email.api-tests.js

```diff
@@ -0,0 +1,74 @@
+const withValidateUsernameAndEmailTests = ({ createRequestBodyWithUpdatedField, makeRequest }) =>
+  describe('when validating the email and username', () => {
+    const EMAIL_ERROR = { text: 'Enter an email address in the correct format, for example, name@example.com' };
+    const USERNAME_AND_EMAIL_MUST_MATCH_ERROR = { text: 'Username and email must match' };
+    const A_DIFFERENT_EMAIL_ADDRESS = 'ADifferentEmailAddress@ukexportfinance.gov.uk';
+    const AN_INVALID_EMAIL_ADDRESS = 'notAValidEmailAddress';
+    const AN_EMPTY_STRING = '';
+    const AN_OBJECT = { aField: 'AString' };
+    const AN_EMPTY_OBJECT = {};
+
+    const errorTestCases = [
+      {
+        description: `is a different email address`,
+        valueToSetField: A_DIFFERENT_EMAIL_ADDRESS,
+        expectedErrorForEmailTests: USERNAME_AND_EMAIL_MUST_MATCH_ERROR.text,
+        expectedErrorForUsernameTests: USERNAME_AND_EMAIL_MUST_MATCH_ERROR.text,
+      },
+      {
+        description: `is an invalid email address`,
+        valueToSetField: AN_INVALID_EMAIL_ADDRESS,
+        expectedErrorForEmailTests: EMAIL_ERROR.text,
+        expectedErrorForUsernameTests: USERNAME_AND_EMAIL_MUST_MATCH_ERROR.text,
+      },
+      {
+        description: `is an empty string`,
+        valueToSetField: AN_EMPTY_STRING,
+        expectedErrorForEmailTests: EMAIL_ERROR.text,
+        expectedErrorForUsernameTests: USERNAME_AND_EMAIL_MUST_MATCH_ERROR.text,
+      },
+      {
+        description: `is an empty object`,
+        valueToSetField: AN_EMPTY_OBJECT,
+        expectedErrorForEmailTests: EMAIL_ERROR.text,
+        expectedErrorForUsernameTests: USERNAME_AND_EMAIL_MUST_MATCH_ERROR.text,
+      },
+      {
+        description: `is an object`,
+        valueToSetField: AN_OBJECT,
+        expectedErrorForEmailTests: EMAIL_ERROR.text,
+        expectedErrorForUsernameTests: USERNAME_AND_EMAIL_MUST_MATCH_ERROR.text,
+      },
+    ];
+    describe('when validating the email', () => {
```

Comment:
Yes this could technically be refactored into just one "describe.each" but as this is a test case, for simplicity of reading I haven't